src/@utils/Wave.ts

```diff
@@ -0,0 +1,174 @@
+import { computeControlPoints } from './bezier-spline'
+import { randomIntFromInterval } from './numbers'
+
+export class Point {
+  x: number
+  y: number
+
+  constructor(x: number, y: number) {
+    this.x = Math.floor(x)
+    this.y = Math.floor(y)
+  }
+}
+
+export interface WaveProperties {
+  width?: number
+  height?: number
+  color?: string
+  fill?: boolean
+  layerCount?: number
+  pointsPerLayer?: number
+  variance?: number
+  maxOpacity?: number
+  opacitySteps?: number
+}
+
+export class SvgWaves {
+  properties: WaveProperties
+  layers: Point[][]
+
+  static xmlns = 'http://www.w3.org/2000/svg'
+  // violet, pink, grey-light
+  static colors = ['#e000cf', '#ff4092', '#8b98a9']
+
+  static getProps(): WaveProperties {
+    return {
+      width: 99,
+      height: 99,
+      color:
+        SvgWaves.colors[randomIntFromInterval(0, SvgWaves.colors.length - 1)],
+      fill: randomIntFromInterval(0, 1) > 0,
+      layerCount: randomIntFromInterval(2, 5),
+      pointsPerLayer: randomIntFromInterval(3, 7),
+      variance: Math.random() * 0.5 + 0.3, // 0.3 - 0.8
+      maxOpacity: 255, // 0xff
+      opacitySteps: 68 // 0x44
```

Comment:
This was thought as a helper to understand where those numbers come from, since opacity values are translated back into hex strings in the different svg paths again.
E.g.: `fill="#ff0000ff`